### PR TITLE
ScrollViewer Improvement

### PR DESCRIPTION
**Pull request by AmRo045**
`IsAutoHideEnabled` and `CornerRectangleVisibility` properties:
**[Preview](https://www.dropbox.com/s/aqpof4pdecqnti1/ScrollViewerPreview.mp4?dl=0)**

`ShowSeparators` property:
ShowSeparators

### CODE REVIEW

File: MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml [hunk 1]
```diff
@@ -10,15 +12,96 @@
...
+                ViewportSize="{TemplateBinding ViewportWidth}" />
         </Grid>
+        <ControlTemplate.Triggers>
+            <Trigger Property="wpf:ScrollViewerAssist.CornerRectangleVisibility" Value="Collapsed">
+                <Setter TargetName="PART_Corner" Property="Visibility" Value="Collapsed"/>
+            </Trigger>
+
+            <Trigger Property="wpf:ScrollViewerAssist.CornerRectangleVisibility" Value="Hidden">
+                <Setter TargetName="PART_Corner" Property="Visibility" Value="Hidden"/>
+            </Trigger>
```
Comment (Keboo) [inline MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml]: Could these be moved into a TemplateBinding on the `PART_Corner`?

File: MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml [hunk 2]
```diff
@@ -10,15 +12,96 @@
...
+                ContentTemplate="{TemplateBinding ContentTemplate}"
+                Content="{TemplateBinding Content}"
+                Grid.Column="0" 
+                Margin="{TemplateBinding Padding}"
+                Grid.Row="0"/>
+
+            <ScrollBar 
+                x:Name="PART_VerticalScrollBar" 
+                AutomationProperties.AutomationId="VerticalScrollBar"
+                Cursor="Arrow"
```
Comment (Keboo) [inline MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml]: Is it necessary to set the Cursor on the Scroll bar?

File: MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml [hunk 3]
```diff
@@ -10,15 +12,96 @@
...
+            </Trigger>
+
+            <Trigger Property="wpf:ScrollViewerAssist.CornerRectangleVisibility" Value="Hidden">
+                <Setter TargetName="PART_Corner" Property="Visibility" Value="Hidden"/>
+            </Trigger>
+
+            <MultiTrigger>
+                <MultiTrigger.Conditions>
+                    <Condition Property="wpf:ScrollViewerAssist.IsAutoHideEnabled" Value="True" />                    
+                    <Condition Property="IsMouseOver" Value="True" />
```
Comment (Keboo) [inline MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml]: I am thinking we should probably add one more condition and separate out the `PART_VerticalScrollBar` and `PART_HorizontalScrollBar` into their own MultiTriggers.
```
<Condition Property="ComputedVerticalScrollBarVisibility" Value="Visible" />
```
That should eliminate the need for the Triggers below that are checking for Disabled and Hidden. It also allows for individually controlling which scrollbars get hidden/shown.

File: MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml (hunk 1 above)
Comment (AmRo045) [reply MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml]: Yes it's better option for setting CornerRectangleVisibility.

File: MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml (hunk 2 above)
Comment (AmRo045) [reply MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml]: I think we don't need set Cursor for scrollbar. So i'll remove this property.

File: MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml (hunk 3 above)
Comment (AmRo045) [reply MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml]: Thank you for the recommendations

**Comment by Keboo** [comment]
@AmRo045 I made one change to move some of the behavior out of the attached property and into triggers on the template. I believe this is a little cleaner since it wont leave event handlers around that might need to get cleaned up. Thoughts?

**Comment by AmRo045** [comment]
@Keboo That's true. Your changes make the code more clean and understandable. Thank you

**Pull request merged**